pkg/sql/streaming.go

```diff
@@ -0,0 +1,918 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
```

Comment:
I'm surprised to find all this new logic here. Is it just copied from `streamingest`? This seems weird to me. Can you keep it there and just call that code from the sql package?

Reply:
I changed it to use those in `streamingest` and `streamproducer`, but directly calling them from `sql` caused this loop
```
.-> //pkg/sql:sql (d5981f1a3d4a46828dfe84289c55e9bfde67ac814b3f9fd2987adc5d9089163d)
|   //pkg/ccl/streamingccl/streamproducer:streamproducer (d5981f1a3d4a46828dfe84289c55e9bfde67ac814b3f9fd2987adc5d9089163d)
`-- //pkg/sql:sql (d5981f1a3d4a46828dfe84289c55e9bfde67ac814b3f9fd2987adc5d9089163d)
```
I made the 3rd commit to break `sql`'s dependency on `streamproducer` with injection. Not very sure if it's correct though...

Reply:
And here's an ugly part -- i passed different interfaces for planner as a parameter for these functions -- e.g. for `CompleteStreamIngestion()` it's `sql.JobExecContext`, for `StartReplicationStream()` it's `sql.PlanHookState`. It's because I had to adapt to the different methods of `*sql.planner` called within them.